packages/clerk-js/src/core/clerk.ts

```diff
@@ -948,7 +952,36 @@ export default class Clerk implements ClerkInterface {
     this.#componentControls?.updateProps(props);
   };
 
-  #loadInStandardBrowser = async (): Promise<void> => {
+  #redirectToSyncWithPrimary = () => {
+    const q = new URLSearchParams({
+      redirect_url: `${window.location.href}?synced=true`,
+    });
+    window.location.replace(new URL(`/v1/client/sync?${q.toString()}`, `https://${this.domain}`).toString());
+  };
+
+  #stripSyncedQueryParam = (q: URLSearchParams) => {
+    q.delete('synced');
+    const queryString = q.toString() ? `?${q.toString()}` : '';
+    window.history.replaceState({}, '', `${window.location.pathname}${queryString}`);
+  };
+
+  #shouldSyncWithMain = () => {
+    const q = new URLSearchParams(window.location.search);
+    if (q.get('synced') !== 'true') {
+      this.#redirectToSyncWithPrimary();
+      return true;
+    }
+    this.#stripSyncedQueryParam(q);
+    return false;
+  };
+
+  #loadInStandardBrowser = async (): Promise<boolean> => {
+    if (this.#options.isSatellite) {
+      if (this.#shouldSyncWithMain()) {
```

Comment:
🔧 The name of this function makes the reader think that it will only do a check and return a boolean. In reality, it also does the redirection and strips the query string parameter. 

I'd suggest to move this functionality inside the #loadInStandardBrowser method, as it will be easier to comprehend the flow.

Reply:
Fixed [here](https://github.com/clerkinc/javascript/pull/727).